scripts/swupd-completion.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+# This script creates a _swupd function to be used by
+# completion built-in bash command to complete swup
+# It reads options and subcommands from swup --help
+# Also it reads options for each subcommand recursively
+
+
+SCRIPTNAME="swupd.bash"
+COMPLETIONFUNCTION="_swupd"
+SWUPDCOMMAND="./swupd"
+
+optionsfromhelp=""
+subcommandsfromhelp=()
+optionsfromsubcommand=()
+
+#Checks if swupd commad exists in current shell
+if [ -a $SWUPDCOMMAND ] ; then
+    currenttype=""
+    nexttype=""
+    array=()
+    #reads "swupd --help" line by line
+    while read line
+    do
+	#For each line checks in what section is
+	if [[ $line == "Help Options:" ]]; then
+	    nexttype="options"
+	fi
+	if [[ $line == "Subcommands:" ]]; then
+	    nexttype="subcommand"
+	fi
+	if [[ $line == "" ]]; then
+	    currenttype=""
+	fi
+	case $currenttype in
+	    #In case of subcommand it inserts first word of line in an array
+	    "subcommand" )
+		array=( $line )
+		subcommandsfromhelp+=(${array[0]}) ;;
+	    #In case of options it reads short and long options
+	    "options" )
+		array=( $line )
+		optionsfromhelp="${optionsfromhelp} ${array[0]%?} ${array[1]}" ;;
+	esac
+	if [[ -n $nexttype ]]; then
+	    currenttype=$nexttype
+	    nexttype=""
+	fi
+    done < <($SWUPDCOMMAND --help)
+else
+    echo "Error: No swupd found"
+    exit 1
+fi
+
+#Now it is time to read options for each subcommand
+for i in "${subcommandsfromhelp[@]}"; do
+    substring=""
+    currenttype=""
+    nexttype=""
+    array=()
+    #Reads "swupd subcommand --help" line by line
+    while read line
+    do
+	# Check when is inside of help options to start read them
+	if [[ $line == "Help Options:" || $line == "Application Options:" ]]; then
+	    nexttype="options"
+	fi
+	if [[ $line == "" ]]; then
+	    currenttype=""
+	fi
+	case $currenttype in
```

Comment:
Similarly for this case block, I think there should be a case for the empty string.